bayesian/api/api_v2.py

```diff
@@ -172,6 +178,90 @@ def get(ecosystem, package, version):
         raise HTTPError(404, msg)
 
 
+class StackAnalysesApi(Resource):
+    """Implements stack analysis REST APIs.
+
+    Implements /api/v2/stack-analyses REST APIs for POST and GET calls.
+    """
+
+    method_decorators = [login_required]
+
+    @staticmethod
+    def get(external_request_id):
+        """Handle /api/v2/stack-analyses GET REST API."""
+        current_app.logger.debug("GET request_id: {}".format(external_request_id))
+
+        # 1. Set stack analyses object
+        sa = StackAnalyses()
+
+        # 2. Read request data for the request
+        db_result = sa.get_request_data(external_request_id)
+        if db_result is None:
+            error_message = "Invalid request ID '{}'.".format(external_request_id)
+            current_app.logger.error(error_message)
+            raise HTTPError(404, error=error_message)
+
+        # 3. Read stack results for the request
+        stack_result = sa.get_stack_result(external_request_id)
+
+        # 4. Read recommendation data for the request
+        recm_data = sa.get_recommendation_data(external_request_id)
+
+        # 5. If stack or reco is missing, either we are in progress or we have timedout.
+        if stack_result is None or recm_data is None:
+            # If the response is not ready and the timeout period is over, send error 408
+            if request_timed_out(db_result):
+                error_message = "Stack analysis request {} has timed out. Please retry " \
+                                "with a new analysis.".format(external_request_id)
+                current_app.logger.error(error_message)
+                raise HTTPError(408, error=error_message)
+            else:
+                error_message = "Analysis for request ID '{}' is in progress".format(
+                    external_request_id)
+                current_app.logger.error(error_message)
+                return {'error': error_message}, 202
+
+        # 6. Some rare case when workers have not updated the result
+        if stack_result == -1 and recm_data == -1:
+            error_message = "Worker result for request ID '{}' doesn't exist yet".format(
+                external_request_id)
+            current_app.logger.error(error_message)
+            raise HTTPError(404, error=error_message)
```

Comment:
Does it make sense to send this error message to the user? Also I think the return code should in the 500 range.

Reply:
This is 404, as given request_id is not valid. So wrong input from user. It makes sense to send this kind of error to user, as they can retry with correct input. More over this is same behaviour from V1.

Reply:
if it is wrong request id, then it would have been handled in the previous condition as no row would be returned from RDS, this seems like an error on our side.

Reply:
I am also not sure when response will be -1 from database, I have added this check based on V1 flow, I assume that there might be a strong reason behind this.